Frameworks/Foundation/NSInvocation.mm

```diff
@@ -160,55 +180,41 @@ - (void)retainArguments {
  @Status Interoperable
 */
 - (void)getReturnValue:(void*)buf {
-    char* type = (char*)[_methodSignature methodReturnType];
-    int length = objc_sizeof_type(type);
-
-    memcpy(buf, returnValue, length);
+    memcpy(buf, _returnValue, _returnLength);
 }
 
 /**
  @Status Interoperable
 */
 - (void)setReturnValue:(void*)buf {
-    char* type = (char*)[_methodSignature methodReturnType];
-    int length = objc_sizeof_type(type);
-
-    if (returnValue) {
-        IwFree(returnValue);
-    }
-
-    returnValue = (char*)IwMalloc(length);
-
-    memcpy(returnValue, buf, length);
+    memcpy(_returnValue, buf, _returnLength);
 }
 
 /**
  @Status Interoperable
 */
 - (void)dealloc {
-    DWORD numArgs = [_methodSignature numberOfArguments];
+    // Release retained/string-copied arguments
+    if (_retainArguments) {
+        auto numArgs = [_methodSignature numberOfArguments];
 
-    //  release arguments
-    if (retainArguments) {
         for (unsigned int i = 0; i < numArgs; i++) {
-            char* type = (char*)[_methodSignature getArgumentTypeAtIndex:i];
-
-            if (args[i] != NULL) {
-                if (strcmp(type, "@") == 0) {
-                    id arg = *((id*)args[i]);
-                    [arg release];
-                }
+            auto type = [_methodSignature getArgumentTypeAtIndex:i];
+            if (type[0] == '@') {
```

Comment:
I'm worried it'd add a little more overhead than I'm comfortable with; what were you thinking it would look like/

---
In reply to: [76333507](https://github.com/Microsoft/WinObjC/pull/903#discussion_r76333507) [](ancestors = 76333507)